tests/main/lxd-mount-units/task.yaml

```
@@ -32,6 +48,18 @@ execute: |
     retry -n 5 --wait 1 sh -c 'lxd.lxc exec ubuntu -- sudo --user ubuntu --login ls /snap/core18/current'
     retry -n 5 --wait 1 sh -c 'lxd.lxc exec ubuntu -- sudo --user ubuntu --login mount | grep core18| grep allow_other'
 
+    echo "Check that snapfuse mount options of generated units match those generated by snapd"
+    # the only way to get mount options used by snapd is to install a new snap
+    retry -n 5 --wait 1 sh -c 'lxd.lxc exec ubuntu snap install hello-world'
+    REFERENCE_MOUNT_OPTS=$(lxd.lxc exec ubuntu -- find /etc/systemd/system -maxdepth 1 -name 'snap-hello*' -exec cat {} \; | grep Options)
```

Maybe we can make it more specific (assuming `\\x2d` works):
```suggestion
    REFERENCE_MOUNT_OPTS=$(lxd.lxc exec ubuntu -- find /etc/systemd/system -maxdepth 1 -name 'snap-hell\\x2d-world-*.mount' -exec cat {} \; | grep Options)
```

However, rather than looking at the file, shouldn't we check what the actual mount unit has? That would be the following snippet:
```
$ systemctl show --property=Options snap-hello\*world-\*.mount
Options=ro,nodev,relatime,x-gdu.hide
```

---

Thanks, I think it's a nice idea. I'd however add this in addition to the above checks, because we want to also catch a potential change where options written by snapd-generator & snapd disagree.

---

As for the being specific, I'd like to avoid tying it to a specific revision (and in this test we will only have a single revision installed).

---

Hmm, actually I'm not sure about the usability of systemctl show --property inside the container, it gives some more options: `Options=ro,nodev,relatime,user_id=0,group_id=0,allow_other`

Since allow_other is explicitely checked above in the test via `mount ... | MATCH`, I'm not sure it's worth checking property for the same information.

---

Interesting, user_id and group_id are fuse options too. Maybe the check above should verify all of those being set? Anyways, it looks like something for a followup.